pkg/ccl/backupccl/key_rewriter.go

```diff
@@ -258,23 +258,33 @@ func (kr *KeyRewriter) RewriteKey(key []byte) ([]byte, bool, error) {
 	return rekeyed, ok, err
 }
 
-// rewriteTableKey rewrites the table IDs in the key.
-// It assumes that any tenant ID has been stripped from the key so it operates
-// with the system codec. It is the responsibility of the caller to either
-// remap, or re-prepend any required tenant prefix.
-func (kr *KeyRewriter) rewriteTableKey(key []byte) ([]byte, bool, error) {
+// ErrImportingKeyError indicates the current key is apart of an in-progress import
+var ErrImportingKeyError = errors.New("Importing Key")
+
+// checkAndRewriteTableKey rewrites the table IDs in the key. It assumes that
+// any tenant ID has been stripped from the key so it operates with the system
+// codec. It is the responsibility of the caller to either remap, or re-prepend
+// any required tenant prefix.
+func (kr *KeyRewriter) checkAndRewriteTableKey(key []byte, wallTime int64) ([]byte, bool, error) {
 	// Fetch the original table ID for descriptor lookup. Ignore errors because
 	// they will be caught later on if tableID isn't in descs or kr doesn't
 	// perform a rewrite.
 	_, tableID, _ := keys.SystemSQLCodec.DecodeTablePrefix(key)
+
+	desc := kr.descs[descpb.ID(tableID)]
+	if desc == nil {
+		return nil, false, errors.Errorf("missing descriptor for table %d", tableID)
+	}
+
+	// Elide Importing Keys
```

Comment:
Lets add more to this comment too, in particular the timestamp comparison and why its what we want in a restore.